syntax = "proto3";

import "google/protobuf/empty.proto";

message Address {
  int32 ip = 1;
  int32 port = 2;
}

message SearchMessage {
  Address addr = 1; // Address of file requester
  string fileName = 2;
  string identifier = 3;
}

message SearchResponse {
  Address addr = 1; // Address of node having the file
  string identifier = 2;
}

message File {
    string name = 1;
    string data = 2;
}

service PeerToPeer {
  rpc SearchFile(SearchMessage) returns (google.protobuf.Empty) {}
  rpc FoundFile(SearchResponse) returns (google.protobuf.Empty) {}
}

message NeighbourList {
  repeated Address neighbours = 1;
}

service PeerMaster {
  rpc PeerJoined(Address) returns (google.protobuf.Empty) {}
  rpc GetNeighbours(Address) returns (NeighbourList) {}
}